# Python Intermedio.

In [12]:
# image_rotator.py

from PIL import Image # importa el módulo Image de la librería PIL

# carga una imagen llamada 'tripleten_logo.jpeg'.
im = Image.open('tripleten_logo.png')

# obtén el tamaño de la imagen mediante el atributo .size y muéstralo
print(im.size)

# gira la imagen 90 grados en sentido contrario a las agujas del reloj
rotated = im.rotate(90)

# guarda la imagen girada
rotated.save('rotated.png')

(1110, 694)


**En este punto python ha rotado una imagen en 90 grados. EL resultado indica el 1110 (ancho) y 694 (alto).**

### **Argumentos**

Los argumentos de línea de comandos son similares a los argumentos de una función en Python. Los pasamos a nuestro script posicionalmente incluyéndolos en nuestra sentencia de ejecución de línea de comandos. Los valores de los argumentos pasados se usan dentro del script.

In [13]:
# image_rotator.py

from PIL import Image
import argparse  # importa el módulo argparse

# inicializa el analizador sintáctico
parser = argparse.ArgumentParser()

# agrega argumentos con sus nombres correspondientes
parser.add_argument("input_file")
parser.add_argument("output_file")
parser.add_argument("angle", type=int)

# analiza los argumentos
args = parser.parse_args()

# carga una imagen del argumento input_file
im = Image.open(args.input_file)

# muestra el tamaño de la imagen
print(im.size)

# gira la imagen en un ángulo proporcionado como argumento
rotated = im.rotate(args.angle)

# guarda la imagen girada usando la ruta de salida de un argumento output_file
rotated.save(args.output_file)

usage: ipykernel_launcher.py [-h] input_file output_file angle
ipykernel_launcher.py: error: the following arguments are required: input_file, output_file, angle


SystemExit: 2

In [18]:
!python image_rotator.py tripleten_logo.png output.png 180


(1110, 694)


# **Manejo de errores**


**Introducción:**

* Los programas de computadora no son tan flexibles como los humanos y se detienen ante errores inesperados.
* Es importante detectar y manejar errores para que el programa no se bloquee y pueda continuar funcionando.

**Mensajes de error:**

* Los mensajes de error nos indican dónde se detectó el error y cuál fue el problema.
* Ejemplo: FileNotFoundError si el nombre del archivo es incorrecto.

**Tipos de excepciones:**

* Python tiene diferentes tipos de excepciones para diferentes tipos de errores.
* Todas las excepciones heredan de la clase Exception.

**Importancia de la detección y manejo de errores:**

* Permite que el programa continúe funcionando a pesar de errores menores.
* Facilita la depuración del programa.
* Puede mejorar la experiencia del usuario al proporcionar información sobre el error y cómo solucionarlo.

**Ejemplo:**

```python
try:
    # Código que puede generar un error
except FileNotFoundError:
    # Manejar el error FileNotFoundError
except TypeError:
    # Manejar el error TypeError
except Exception:
    # Manejar cualquier otro tipo de error
```

**En resumen:**

* La detección y el manejo de errores son esenciales para escribir programas robustos y fáciles de usar en Python.
* La estructura `try-except` te permite capturar diferentes tipos de excepciones y ejecutar código específico para cada una de ellas.

**Consejos:**

* Usa mensajes de error claros y concisos.
* Proporciona información sobre cómo solucionar el error.
* Registra los errores para poder analizarlos posteriormente.

